conan/tools/qbs/qbsprofile.py

```diff
@@ -220,20 +231,25 @@ class QbsProfile(object):
                 {%- if build_variant %}
                 qbs.buildVariant: "{{ build_variant }}"
                 {%- endif %}
-                {%- if architecture %}
+                {%- if not _profile_values_from_setup["qbs.architecture"] and architecture %}
```

Comment:
This might go also in the context definition (The context decides the "architecture" value, the template renders it)